File: AmdarisProject.Angular/src/app/common/models/pagination/pagination-request.ts
import { MatPaginator } from "@angular/material/paginator";
import { SortingRequest } from "./sorting-request";

export class PaginationRequest {
    pageIndex: number;
    pageSize: number;
    sortingRequest: SortingRequest;

    constructor(matPaginator: MatPaginator, sortingRequest: SortingRequest) {
        this.pageIndex = matPaginator.pageIndex;
        this.pageSize = matPaginator.pageSize;
        this.sortingRequest = sortingRequest;
    }
}

File: AmdarisProject.Common/Models/PaginationRequest.cs
namespace AmdarisProject.Common.Models
{
    public class PaginationRequest
    {
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public SortingRequest SortingRequest { get; set; }
        public FilterRequest FilterRequest { get; set; }
    }
}
